src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java

```diff
@@ -217,6 +218,21 @@ public void setupProgrammingExercise_validExercise_created(ExerciseMode mode) th
         assertThat(programmingExerciseRepository.count()).isEqualTo(1);
     }
 
+    @ParameterizedTest
+    @EnumSource(ExerciseMode.class)
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void setupProgrammingExerciseWithoutLocalSetup_validExercise_created(ExerciseMode mode) throws Exception {
+        exercise.setMode(mode);
+        // mockConnectorRequestsForSetup(exercise);
+        final var generatedExercise = request.postWithResponseBody(
+                ProgrammingExerciseSimulationResource.Endpoints.ROOT + ProgrammingExerciseSimulationResource.Endpoints.EXERCISES_SIMULATION, exercise, ProgrammingExercise.class,
+                HttpStatus.CREATED);
+
+        exercise.setId(generatedExercise.getId());
+        assertThat(exercise).isEqualTo(generatedExercise);
```

Comment:
Together with the previous line, this is trivially true. You are setting the id of `exercise` to the id of `generatedExercise` and then you are checking if the ids are equal (`isEqualTo` used the `equals` method of `Exercise` which does nothing more then comparing the ids).

Reply:
i adjusted the test case